centaur/src/main/scala/centaur/api/DaemonizedDefaultThreadFactory.scala

```diff
@@ -0,0 +1,25 @@
+package centaur.api
+
+import java.util.concurrent.ThreadFactory
+import java.util.concurrent.atomic.AtomicInteger
+
+/**
+  * A static version of java.util.concurrent.Executors.DefaultThreadFactory that creates daemon threads that exit when
+  * the application exits.
+  *
+  * java.util.concurrent.Executors.DefaultThreadFactory
+  * http://dev.bizo.com/2014/06/cached-thread-pool-considered-harmlful.html
```

Comment:
Is this URL intended to be the explanation of why this thing exists? If so can you give a brief summary. If not, can you add an explanation of why this is here?

Reply:
Will do. TL;DR: non-daemon threads exit when they want to, not when a user presses control-C or System.exit() is called.